core/new-gui/src/app/dashboard/service/user-project/user-project.service.ts

```diff
@@ -0,0 +1,105 @@
+  private files: ReadonlyArray<DashboardUserFileEntry> = [];
+
+  constructor(private http: HttpClient, private userFileService: UserFileService) { 
+  }
+
+  public retrieveProjectList() : Observable<Project[]>{
+    return this.http.get<Project[]>(`${USER_PROJECT_LIST_URL}`);
+  }
+
+  public retrieveWorkflowsOfProject(pid: number) : Observable<DashboardWorkflowEntry[]>{
+    return this.http.get<DashboardWorkflowEntry[]>(`${WORKFLOW_OF_PROJECT_URL}/` + pid);
+  }
+
+  public retrieveFilesOfProject(pid: number) : Observable<DashboardUserFileEntry[]>{
+    return this.http.get<DashboardUserFileEntry[]>(`${FILE_OF_PROJECT_URL}/` + pid );
+  }
+
+  public getProjectFiles(): ReadonlyArray<DashboardUserFileEntry> {
+    return this.files;
+  }
+
+  public refreshFilesOfProject(pid: number): void {
+    this.retrieveFilesOfProject(pid)
+      .subscribe(files => {
+        this.files = files;
+      });
+  }
+
+  public retrieveProject(pid: number) : Observable<Project>{
+    return this.http.get<Project>(`${PROJECT_BASE_URL}/` + pid);
+  }
+
+  public updateProjectName(pid: number, name: string) : Observable<Response>{
+    return this.http.post<Response>(`${RENAME_PROJECT_URL}/` + pid + "/" + name, {});
+  }
+
+  public deleteProject(pid: number): Observable<Response> {
+    return this.http.delete<Response>(`${DELETE_PROJECT_URL}/` + pid);
+  }
+
+  public createProject(name: string) : Observable<Project>{
+    return this.http.post<Project>(`${CREATE_PROJECT_URL}/` + name, {});
+  }
+
+  public addWorkflowToProject(pid: number, wid: number): Observable<Response>{
+    return this.http.post<Response>(`${ADD_WORKFLOW_TO_PROJECT_URL}/` + pid + "/" + wid, {});
+  }
+
+  public removeWorkflowFromProject(pid: number, wid: number): Observable<Response>{
+    return this.http.delete<Response>(`${DELETE_WORKFLOW_FROM_PROJECT_URL}/` + pid + "/" + wid, {});
+  }
+
+  public addFileToProject(pid: number, fid: number): Observable<Response>{
+    return this.http.post<Response>(`${ADD_FILE_TO_PROJECT_URL}/` + pid + "/" + fid, {});
+  }
+
+  public removeFileFromProject(pid: number, fid: number): Observable<Response>{
+    return this.http.delete<Response>(`${DELETE_FILE_FROM_PROJECT_URL}/` + pid + "/" + fid, {});
+  }
+
+  /**
+   * same as UserFileService"s deleteDashboardUserFileEntry method, except
+   * it is modified to refresh the project"s list of files
+   */
+   public deleteDashboardUserFileEntry(pid: number, targetUserFileEntry: DashboardUserFileEntry): void {
+    this.http
+      .delete<Response>(`${USER_FILE_DELETE_URL}/${targetUserFileEntry.file.name}/${targetUserFileEntry.ownerName}`)
+      .subscribe(
+        () => {
+          this.userFileService.refreshDashboardUserFileEntries();
+          this.refreshFilesOfProject(pid); // refresh files within project
+        },
+        // @ts-ignore // TODO: fix this with notification component
+        (err: unknown) => alert("Cannot delete the file entry: " + err.error)
+      );
+  }
```

Comment:
please avoid making duplicates of methods.

Reply:
will refactor code to unify duplicate logic and components in future PR